sphinx_gallery/binder.py

```diff
@@ -0,0 +1,130 @@
+import shutil as sh
+import os
+
+try:
+    basestring
+except NameError:
+    basestring = str
+    unicode = str
+
+
+def gen_binder_url(fname, binder_conf):
+    """Generate a Binder URL according to the configuration in conf.py.
+
+    Parameters
+    ----------
+    fname: str
+        The path to the `.py` file for which a Binder badge will be generated.
+    binder_conf: dict | None
+        The Binder configuration dictionary. See `gen_binder_rst` for details.
+
+    Returns
+    -------
+    binder_url : str
+        A URL that can be used to direct the user to the live Binder
+        environment.
+    """
+    # Build the URL
+    binder_fpath = '_downloads/{}'.format(fname.replace('.py', '.ipynb'))
+    binder_url = binder_conf['url']
+    binder_url = '/'.join([binder_conf['url'],
+                           'v2', 'gh',
+                           binder_conf['org'],
+                           binder_conf['repo'],
+                           binder_conf['branch']])
+    binder_url += '?filepath={}'.format(binder_fpath)
+    return binder_url
+
+
+def gen_binder_rst(fname, binder_conf):
+    """Generate the RST + link for the Binder badge.
+
+    Parameters
+    ----------
+    fname: str
+        The path to the `.py` file for which a Binder badge will be generated.
+    binder_conf: dict | None
+        If a dictionary it must have the following keys:
+
+        'url': The URL of the BinderHub instance that's running a Binder
+            service.
+        'org': The GitHub organization to which the documentation will be
+            pushed.
+        'repo': The GitHub repository to which the documentation will be
+            pushed.
+        'branch': The Git branch on which the documentation exists (e.g.,
+            gh-pages).
+        'dependencies': A list of paths to dependency files that match the
+            Binderspec.
+
+    Returns
+    -------
+    rst : str
+        The reStructuredText for the Binder badge that links to this file.
+    """
+    binder_url = gen_binder_url(fname, binder_conf)
+    rst = (".. figure:: http://mybinder.org/badge.svg\n"
+           "      :target: {}\n").format(
+        binder_url)
+    rst += "      :width: 150 px\n      :figclass: binder-badge\n\n"
+    return rst
+
+
+def copy_binder_reqs(app):
+    """Copy Binder requirements files to a "binder" folder in the docs."""
+    binder_conf = app.config.sphinx_gallery_conf['binder']
+    path_reqs = binder_conf.get('dependencies')
+
+    binder_folder = os.path.join(app.builder.outdir, 'binder')
```

Comment:
This may be the reason dependencies are not installed in my binder. According to the binder docs or examples it seems to look for the files in root. I'm I missing something?

Reply:
Discard this comment. This is not the problem.